src/pages/methods/workflow-details/WorkflowWrapper.test.tsx

```diff
@@ -407,6 +415,75 @@ describe('workflows container', () => {
     expect(goToPath).toHaveBeenCalledWith('workflows');
   });
 
+  it('renders the clone snapshot modal when the corresponding button is pressed', async () => {
+    // Arrange
+    mockAjax();
+
+    // set the user's email
+    jest.spyOn(userStore, 'get').mockImplementation(jest.fn().mockReturnValue(mockUserState('hello@world.org')));
+
+    const user: UserEvent = userEvent.setup();
+
+    // Act
+    await act(async () => {
+      render(
+        <WorkflowsContainer
+          namespace={mockSnapshot.namespace}
+          name={mockSnapshot.name}
+          snapshotId={`${mockSnapshot.snapshotId}`}
+          tabName='dashboard'
+        />
+      );
+    });
+
+    await user.click(screen.getByRole('button', { name: 'Snapshot action menu' }));
+    await user.click(screen.getByRole('button', { name: 'Clone snapshot' }));
+
+    const dialog = screen.getByRole('dialog', { name: /clone snapshot/i });
+
+    // Assert
+    expect(dialog).toBeInTheDocument();
+    expect(within(dialog).getByRole('textbox', { name: 'Namespace *' })).toHaveDisplayValue('');
+    expect(within(dialog).getByRole('textbox', { name: 'Name *' })).toHaveDisplayValue('testname_copy');
+    expect(within(dialog).getByRole('textbox', { name: 'Documentation' })).toHaveDisplayValue('mock documentation');
+    expect(within(dialog).getByRole('textbox', { name: 'Synopsis (80 characters max)' })).toHaveDisplayValue('');
+    expect(within(dialog).getByRole('textbox', { name: 'Snapshot comment' })).toHaveDisplayValue(
+      'mock snapshot comment'
+    );
+    expect(within(dialog).getByTestId('wdl editor')).toHaveDisplayValue(mockSnapshot.payload.toString());
+  });
+
```

Comment:
Do you think that test case would be covered by [this test](https://github.com/DataBiosphere/terra-ui/pull/5162/files#diff-9bb324f9586874f756a693e1e6561ae985a74cb72ce14f47ab4dd64eff3b7582R482-R540)? This test checks that the postMethodProvider has been called with correct values on submit.

Reply:
That test is for the modal component itself, and as far as I can tell doesn't really test anything not already tested by the existing tests in that file (which makes sense since the only changes to the modal component were naming changes). The test I am suggesting would be for the WorkflowWrapper, to make sure that the right functions are called by the clone modal specifically as used in the workflow wrapper. (For `onSuccess`, for example: for the create workflow modal, the tests for WorkflowModal check that whatever function was given for `onSuccess` will be called, and the tests for WorkflowList check that specifically `Nav.goToPath` is called by the actual modal used on the workflow list page.)

Reply:
This test should probably cover `postMethodProvider` and `onSuccess`, but SonarCloud shows that the real `onSuccess` code is never being run in tests.

Reply:
I have added the suggested test.